pilot/cmd/pilot-agent/config/config.go

```diff
@@ -85,6 +96,20 @@ func ConstructProxyConfig(meshConfigFile, serviceCluster, proxyConfigEnv string,
 	return applyAnnotations(proxyConfig, annotations), nil
 }
 
+// determineConcurrencyOption determines the correct setting for --concurrency based on CPU limits
+func determineConcurrencyOption() *wrapperspb.Int32Value {
+	// If limit is set, us that
+	// The format in the file is a plain integer. `100` in the file is equal to `100m` (based on `divisor: 1m`
+	// in the pod spec).
+	// With the resource setting, we round up to single integer number; for example, if we have a 500m limit
+	// the pod will get concurrency=1. With 6500m, it will get concurrency=7.
+	limit, err := readPodCPULimits()
+	if err == nil && limit > 0 {
```

Comment:
The old logic used requests and limit. Its not clear if or why only limits is sufficient now: https://github.com/istio/istio/pull/32928/files. Can you expand?